src/coreclr/src/jit/valuenum.cpp

```diff
@@ -6687,7 +6683,16 @@ void Compiler::fgValueNumberBlockAssignment(GenTree* tree)
                         {
                             var_types     indType            = lclVarTree->TypeGet();
                             ValueNum      fieldSeqVN         = srcAddrFuncApp.m_args[0];
+                            // It could return a sequence that ends with a pseudo fields that does not make sense for me.
```

Comment:
for some static vars we are getting field seq that ends with a pseudo field, not sure why. It could be another issue.
[x86.failure.txt](https://github.com/dotnet/runtime/files/5291458/x86.failure.txt)

Reply:
I am looking for a better way to handle this.

Reply:
Pseudo fields are things like the first element of an array